git/client.go

```diff
@@ -336,14 +336,28 @@ func (c *Client) DeleteLocalTag(ctx context.Context, tag string) error {
 }
 
 func (c *Client) DeleteLocalBranch(ctx context.Context, branch string) error {
-	args := []string{"branch", "-D", branch}
-	cmd, err := c.Command(ctx, args...)
-	if err != nil {
-		return err
+	{
+		args := []string{"branch", "-D", branch}
+		cmd, err := c.Command(ctx, args...)
+		if err != nil {
+			return err
+		}
+		_, err = cmd.Output()
+		if err != nil {
+			return err
+		}
 	}
-	_, err = cmd.Output()
-	if err != nil {
-		return err
+	{
+		track := fmt.Sprintf("origin/%s", branch)
```

Comment:
Hmmm, I think we're going to need to inject something here. We can't know for sure that their remote is called `origin`. @samcoe can you advise?

Reply:
Yeah we will want to make sure that we have the correct remote name. Perhaps that is a n indication that we should create a new method `DeleteLocalTrackingBranch` that takes in the remote name and the remote branch name. 

Or we reverse course and go with `git push --delete` since it only requires the branch name and automatically figures out the tracking branch to delete based on the push URL.